### ISSUE TITLE
Why using `log_softmax` instead of `softmax`?

### ISSUE THREAD

**Issue by nguyenvulong** [yoshitomo-matsubara/torchdistill#233]
Same question has been asked [here ](https://github.com/HobbitLong/RepDistiller/issues/52) and [here ](https://github.com/AberHu/Knowledge-Distillation-Zoo/issues/12). These repositories (I think you already know them) are other attempts to implement knowledge distillation algorithms.

Could you please explain why it used `log_softmax` instead of `softmax`?

https://github.com/yoshitomo-matsubara/torchdistill/blob/993ee945d783ea0d8bd007da43c5b03d5a426bec/torchdistill/losses/single.py#L99-L106

**Comment by yoshitomo-matsubara** [comment]
Hi @nguyenvulong 

See [KLDivLoss in PyTorch document](https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss).
> To avoid underflow issues when computing this quantity, this loss expects the argument input in the log-space. The argument target may also be provided in the log-space if log_target= True.


Also, please use **Discussions** above (instead of **Issues**) for questions.
As explained [here](https://github.com/yoshitomo-matsubara/torchdistill#issues--questions--requests), I want to keep **Issues** mainly for bug reports.
